hack/generated/controllers/crd_managedcluster_test.go

```diff
@@ -0,0 +1,152 @@
+
+	cluster := &aks.ManagedCluster{
+		ObjectMeta: tc.MakeObjectMeta("mc"),
+		Spec: aks.ManagedClusters_Spec{
+			Location:  tc.AzureRegion,
+			Owner:     testcommon.AsOwner(rg.ObjectMeta),
+			DnsPrefix: to.StringPtr("aso"),
+			AgentPoolProfiles: []aks.ManagedClusterAgentPoolProfile{
+				{
+					Name:   "ap1",
+					Count:  to.IntPtr(1),
+					VmSize: to.StringPtr("Standard_DS2_v2"),
+					OsType: &osType,
+					Mode:   &agentPoolMode,
+				},
+			},
+			LinuxProfile: &aks.ContainerServiceLinuxProfile{
+				AdminUsername: adminUsername,
+				Ssh: aks.ContainerServiceSshConfiguration{
+					PublicKeys: []aks.ContainerServiceSshPublicKey{
+						{
+							KeyData: *sshPublicKey,
+						},
+					},
+				},
+			},
+			Identity: &aks.ManagedClusterIdentity{
+				Type: &identityKind,
+			},
+		},
+	}
+
+	tc.CreateResourceAndWait(cluster)
+
+	tc.Expect(cluster.Status.Id).ToNot(BeNil())
+	armId := *cluster.Status.Id
+
+	// Perform a simple patch
+	skuName := aks.ManagedClusterSKUNameBasic
+	skuTier := aks.ManagedClusterSKUTierPaid
+	old := cluster.DeepCopy()
+	cluster.Spec.Sku = &aks.ManagedClusterSKU{
+		Name: &skuName,
+		Tier: &skuTier,
+	}
+	tc.Patch(old, cluster)
+
+	objectKey := client.ObjectKeyFromObject(cluster)
+
+	// ensure state got updated in Azure
+	tc.Eventually(func() bool {
+		updated := &aks.ManagedCluster{}
+		tc.GetResource(objectKey, updated)
+		return updated.Status.Sku != nil && *updated.Status.Sku.Name == aks.ManagedClusterSKUStatusNameBasic && *updated.Status.Sku.Tier == aks.ManagedClusterSKUStatusTierPaid
+	}).Should(BeTrue())
+
+	// Run sub tests
+	tc.RunParallelSubtests(
+		testcommon.Subtest{
+			Name: "AKS AgentPool CRUD",
+			Test: func(testContext testcommon.KubePerTestContext) {
+				ManagedCluster_AgentPool_CRUD(testContext, cluster.ObjectMeta)
+			},
+		},
+	)
+
+	tc.DeleteResourceAndWait(cluster)
+
+	// Ensure that the resource group was really deleted in Azure
+	exists, retryAfter, err := tc.AzureClient.HeadResource(tc.Ctx, armId, string(aks.ManagedClustersSpecAPIVersion20210501))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func ManagedCluster_AgentPool_CRUD(tc testcommon.KubePerTestContext, cluster metav1.ObjectMeta) {
+	osType := aks.ManagedClusterAgentPoolProfilePropertiesOsTypeLinux
+	agentPoolMode := aks.ManagedClusterAgentPoolProfilePropertiesModeSystem
+
+	agentPool := &aks.ManagedClustersAgentPool{
+		ObjectMeta: tc.MakeObjectMetaWithName("ap2"),
+		Spec: aks.ManagedClustersAgentPools_Spec{
+			Owner:  testcommon.AsOwner(cluster),
+			Count:  to.IntPtr(1),
+			VmSize: to.StringPtr("Standard_DS2_v2"),
+			OsType: &osType,
+			Mode:   &agentPoolMode,
+		},
+	}
+
+	tc.CreateResourceAndWait(agentPool)
+	// Note: If this test is just too slow to record (it's pretty slow), we can remove this
+	// delete entirely which will speed it up some. Leaving it here for now though as we
+	// need it to test the DELETE path of agent pool.
```

Comment:
You mean we can remove this because we know it'll be cleaned up by deleting the cluster?

Reply:
Yeah, I wanted to make sure this worked in additional to the cluster delete, but we could give up that check to go faster (because then basically all the deletes are happening in parallel)

Reply:
I clarified this comment a bit to make sure that this was clear.